### DISCUSSION TITLE
Stepper: Why is the value of an input element not available in the next step?

### DISCUSSION THREAD

**Discussion by rodja** [zauberzeug/nicegui#2312] [Q&A] [answered]
### Question

The question came up on [Discord](https://discord.com/channels/1089836369431498784/1192738872434638878). Here is a minimal example which surprisingly does not show the name entered in the first step:

```py
from nicegui import ui

with ui.stepper() as stepper:
    with ui.step('Input'):
        name = ui.input('Name', on_change=lambda e: print(e.value))
        with ui.stepper_navigation():
            ui.button('Next', on_click=stepper.next)
    with ui.step('Output'):
        ui.label(f'Hello {name.value}')
        with ui.stepper_navigation():
            ui.button('Back', on_click=stepper.previous)

ui.run()
```

**Comment by falkoschindler** [comment, marked as answer]
The label is created when the stepper is created with all its steps, long before the user inputs any value. Therefore its text contains the initial value.

To update the label text dynamically, you can bind it:
```py
ui.label().bind_text_from(name, 'value', lambda v: f'Hello {v}!')
```
